packages/ui-extensions/src/surfaces/checkout/components/PaymentIcon/PaymentIcon.ts

```diff
@@ -0,0 +1,481 @@
+  | 'nationale-entertainmentcard'
+  | 'podium-cadeaukaart'
+  | 'webshop-giftcard'
+  | 'billease'
+  | 'lku'
+  | 'rietumu'
+  | 'affirm'
+  | 'splitit'
+  | 'zip'
+  | 'humm'
+  | 'synchrony'
+  | 'satisfi'
+  | 'checkout-finance'
+  | 'n26'
+  | 'echelon-financing'
+  | 'acima-leasing'
+  | 'synchrony-pay'
+  | 'truemoney-pay'
+  | 'gmo-postpay'
+  | 'giftcard'
+  | 'siauliu-bankas'
+  | 'medicinos-bankas'
+  | 'epospay'
+  | 'poli'
+  | 'perlasfinance'
+  | 'directpay'
+  | 'latitudepay'
+  | 'genoapay'
+  | 'poppankki'
+  | 'revolut'
+  | 'aktia'
+  | 'saastopankki'
+  | 'spankki'
+  | 'walley'
+  | 'atone'
+  | 'addi'
+  | 'omasp'
+  | 'handelsbanken'
+  | 'alandsbanken'
+  | 'siirto'
+  | 'scalapay'
+  | 'aplazo'
+  | 'paybylink'
+  | 'snap-checkout'
+  | 'fairstone-payments'
+  | 'upi'
+  | 'sveaeramaksu'
+  | 'svea-part-payment'
+  | 'svea-delbetalning'
+  | 'svea-lasku'
+  | 'svea-invoice'
+  | 'svea-faktura'
+  | 'svea-yrityslasku'
+  | 'svea-b2b-invoice'
+  | 'svea-b2b-faktura'
+  | 'pix'
+  | 'wegetfinancing'
+  | 'anyday'
+  | 'cleverpay'
+  | 'payid'
+  | 'uangme'
+  | 'grailpay'
+  | 'payto'
+  | 'zinia'
+  | 'fortiva'
+  | 'fintecture'
+  | 'hello-clever'
+  | 'credit-key'
+  | 'palawa'
+  | 'palawan'
+  | 'kredivo'
+  | 'uob-ez-pay'
+  | 'akulaku-paylater'
+  | 'tandympayment'
+  | 'paytomorrow'
+  | 'bsi'
+  | 'clave-telered'
+  | 'credix'
+  | 'sinpe-movil'
+  | 'laybuy-heart'
+  | 'lydia'
+  | 'atome'
+  | 'creditclick-paynl-version'
+  | 'dropp'
+  | 'cebuana'
+  | 'futurepay-mytab'
+  | 'biercheque-paynl-version'
+  | 'swiftpay'
+  | 'linkpay'
+  | 'riverty-paynl-version'
+  | 'wechat-paynl-version'
+  | 'vvv-cadeaukaart-paynl-version'
+  | 'przelewy24-paynl-version'
+  | 'paysafecard-paynl-version'
+  | 'gezondheidsbon-paynl-version'
+  | 'fashion-giftcard-paynlversion'
+  | 'alipay-paynl-version'
+  | 'aftee'
+  | 'momopay'
+  | 'zalopay'
+  | 'finloup'
+  | 'billink-method'
+  | 'oxxo'
+  | 'spei'
+  | 'bbva-cie'
+  | 'alma'
+  | 'svea-ostukonto'
+  | 'svea-credit-account'
+  | 'biller-paynl-version'
+  | 'swissbilling'
+  | 'bizum'
+  | 'rcbc'
+  | 'inbank'
+  | 'ivy'
+  | 'zoodpay'
+  | 'valu'
+  | 'tabby'
+  | 'cliq'
+  | 'postpay'
+  | 'urpay'
+  | 'hypercash'
+  | 'fawry'
+  | 'spotii'
+  | 'tamara'
+  | 'aqsat'
+  | 'careem-pay'
+  | 'billie'
+  | 'sbpl'
+  | 'coppel-pay'
+  | 'sequra'
+  | 'aeropay'
+  | 'catch-payments'
+  | 'openpay'
+  | 'lbc'
+  | 'atrato'
+  | 'bss'
+  | 'gmo-bank-transfer'
+  | 'trevipay'
+  | 'paypo'
+  | 'younited-pay'
+  | 'tabit'
+  | 'etika'
+  | 'mokka'
+  | 'myfatoorah'
+  | 'meeza'
+  | 'paysafecash'
+  | 'tbi-bank'
+  | 'skrill-digital-wallet'
+  | 'pei'
+  | 'rapid-transfer'
+  | 'neteller'
+  | 'cetelem'
+  | 'paid';
+
+export interface PaymentIconProps {
+  /**
+   * The name of the payment method.
+   *
+   * Check the list of available payment methods [here](https://github.com/activemerchant/payment_icons/tree/master/app/assets/images/payment_icons).
```

Comment:
Just noticed the link. We should point them to the `PaymentMethod` type no?

I don't believe the names of the icons in the the active_merchant repo matches the ones we accept here, so I'd remove this link to avoid any confusion. Developers should only refers to the list of PaymentMethod listed above.

Reply:
Removed the link. Do you know why the [docs](https://shopify-dev.payment-icon.lianne-sit.us.spin.dev/docs/api/checkout-ui-extensions/unstable/components/media/paymenticon#paymenticonprops) only show "string" even though the type is `PaymentMethod | string`?

Reply:
Yep I asked about it in #core-build-learn and they were able to reproduce the issue and logged an issue on their end for it https://github.com/Shopify/generate-docs/issues/217

Reply:
Can this be shipped as is and it'll be fixed eventually or should I wait for their fix? For context, I'm trying to get this component exposed for the 2023-10 release and the cutoff date is next week.

Reply:
Go ahead. The docs can be fixed after, or manually modified if possible.

Reply:
Turns out that because `PaymentMethod` encompasses a list of strings, so as far as typescript is concerned, the type is now just "string" since it can be a list of the PaymentMethod's string or a string which ultimately results in a string. The #core-build-learn team's suggestion was to add the definition of PaymentMethod to the page, which I've done now.